src/ProviderBasedGenerator.php

```diff
@@ -69,6 +69,11 @@ public function generate($name, $parameters = [], $referenceType = self::ABSOLUT
             throw new RouteNotFoundException(sprintf('Route "%s" does not exist.', $name));
         }
 
+        // the parameter must be unset to avoid unexpected generation behaviour
+        if (array_key_exists(RouteObjectInterface::ROUTE_OBJECT, $parameters)) {
+            unset($parameters[RouteObjectInterface::ROUTE_OBJECT]);
```

Comment:
to avoid any possible side effects, i would move this into the `elseif` right after line 67.

Reply:
Good point. I updated the code and use it inside the ``elseif``